lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py

```diff
@@ -0,0 +1,194 @@
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.options = self.params.get('options', dict())
+        self.hosts = []
+        if cluster_name:
+            cluster_obj = self.find_cluster_by_name(cluster_name=cluster_name)
+            if cluster_obj:
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
+            if esxi_host_obj:
+                self.hosts = [esxi_host_obj]
+            else:
+                module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+        self.desired_state = self.params.get('state')
+        self.service_name = self.params.get('service_name')
+        self.results = {}
+
+    def service_ctrl(self):
+        changed = False
+        host_service_state = []
+        for host in self.hosts:
+            actual_service_state = self.check_service_state(host=host, service_name=self.service_name)
+            host_service_system = host.configManager.serviceSystem
+            if host_service_system:
+                service_state = False
+                self.results[host.name] = dict(service_name=self.service_name,
+                                               actual_service_state='running' if actual_service_state else 'stopped',
+                                               desired_service_state=self.desired_state,
+                                               error='',
+                                               )
+                try:
+                    if self.desired_state in ['start', 'present']:
+                        if not actual_service_state:
+                            host_service_system.StartService(id=self.service_name)
+                            service_state = True
+                    elif self.desired_state in ['stop', 'absent']:
+                        if actual_service_state:
+                            host_service_system.StopService(id=self.service_name)
+                            service_state = True
+                    elif self.desired_state == 'restart':
+                        host_service_system.RestartService(id=self.service_name)
+                        service_state = True
+                    host_service_state.append(service_state)
+                    self.results[host.name].update(changed=service_state)
+                except (vim.fault.InvalidState, vim.fault.NotFound, vim.fault.HostConfigFault) as e:
+                    self.results[host.name].update(changed=False,
+                                                   error=to_native(e.msg))
+
+        if any(host_service_state):
+            changed = True
+        self.module.exit_json(changed=changed, results=self.results)
+
+    def check_service_state(self, host, service_name):
```

Comment:
Can be replaced with:
```python
def check_service_state(self, host, service_name):
    host_service_system = host.configManager.serviceSystem
    if host_service_system:
        services = host_service_system.serviceInfo.service
        for service in services:
            if service.key == service_name:
                return service.running

    msg = "Failed to find '%s' service on host system '%s'" % (service_name, host.name)
    cluster_name = self.params.get('cluster_name', None)
    if cluster_name:
        msg += " located on cluster '%s'" % cluster_name
    msg += ", please check if you have specified a valid ESXi service name."
    self.module.fail_json(msg=msg)
```
This way it stops looping through services as soon as it finds the correct one, and if it finds none, it fails

Reply:
OK.